### ISSUE TITLE
Status broken with UDMP on latest firmware (1.8.5)

### ISSUE THREAD

**Issue by adarobin** [paultyng/go-unifi#20]
I have a UDMP running firmware 1.8.5 which includes network controller 6.0.43.  When I attempt to use the terraform provider, I receive a 401 trying to access the `/proxy/network/status` endpoint.

```
Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

unifi_user.truenas: Creating...

Error:   (401 Unauthorized) for GET https://unifi.home.xxxxxxxx.net/proxy/network/status
```

When I attempt to navigate to https://unifi.home.xxxxxxxx.net/api/auth/login, I get the message "Method Not Allowed".  From [this](https://community.ui.com/releases/UniFi-Network-Controller-6-0-43/b28bb453-927a-4dd8-82f7-40af28505510?page=4) forum thread, it looks like the login endpoint has changed back to `/api/login`.

**Comment by adarobin** [comment]
Perhaps have the login function check to see if the response from the login attempt is a 405 and then try to use the other login endpoint if it fails?

**Comment by adarobin** [comment]
So the 405 might be a red herring, that was because I did a GET request to that endpoint, instead of a POST.  I am able to successfully login with curl.

I set up a mitmproxy, and it doesn't look like the Terraform provider is even attempting to log in.
```
   GET https://unifi.home.xxxxxxxx.net/
       ← 200 text/html 357b 53ms
>> GET https://unifi.home.xxxxxxxx.net/proxy/network/status
       ← 401 text/plain 12b 20ms
```

**Comment by adarobin** [comment]
Ahh, I see now. I poked around in the code, and it looks like that endpoint didn't require auth in the past.

```
adam@wintendo:~/network-home$ curl https://unifi.home.xxxxxxxx.net/proxy/network/status && echo
Unauthorized
adam@wintendo:~/network-home$ curl -s -X POST -c /tmp/unificookie -D /tmp/unifiheaders -H "Content-Type: application/json" --data '{"username": "terraform", "password": "redacted"}' https://unifi.home.xxxxxxxx.net/api/auth/login > /dev/null
adam@wintendo:~/network-home$ curl --cookie /tmp/unificookie --cookie-jar /tmp/unificookie -H '$(grep X-CSRF-Token /tmp/unifiheaders)' https://unifi.home.xxxxxxxx.net/proxy/network/status && echo
{"meta":{"rc":"ok","uuid":"15eed773-1721-5f31-9926-89112a62df52"},"data":[]}
```

That endpoint doesn't appear to return much information anymore either. Looks like even if we authenticated for that end point it wouldn't be able to find the version.

Now I'm not sure if this is a UDM thing or a Network Controller 6.0.43 thing.

**Comment by paultyng** [comment]
Argh, yeah I will need to see what home assistant does among others I guess. I unfortunately do not have a UDMP so hard to test/simulate this.

**Comment by paultyng** [comment]
Here is the home assistant issue just for my reference: https://github.com/home-assistant/core/issues/45047

**Comment by adarobin** [comment]
@paultyng Have you done any testing with a regular network controller on 6.0.43 yet? I was able to get login working, but then I hit an issue with the UpdateUser API call. I fixed that as well, but I wonder if its a UDMP specific thing or something with 6.0.43.

I was able to get the terraform provider to work with this commit
https://github.com/adarobin/go-unifi/commit/6c4fe4896e2e232977494a0fceb652a2444b33bd
https://github.com/adarobin/go-unifi/commit/99aefb9abb6beb360e3a20b0a82bb5acf5fcbdcc
